server/telemetry.go

```diff
@@ -3,65 +3,49 @@ package main
 import (
 	"strings"
 
+	"github.com/mattermost/mattermost-plugin-api/experimental/bot/logger"
+	"github.com/mattermost/mattermost-plugin-api/experimental/telemetry"
 	"github.com/mattermost/mattermost-server/v5/model"
-	analytics "github.com/segmentio/analytics-go/v2"
 )
 
 const (
 	NPS_FEEDBACK = "nps_feedback"
 	NPS_SCORE    = "nps_score"
-
-	SEGMENT_KEY = "5xaDYWpjOoCKmJNNKK6fg1DacwZ7ZVZc"
 )
 
 func (p *Plugin) initializeClient() error {
-	client := analytics.New(SEGMENT_KEY)
-
-	if !p.blockSegmentEvents {
-		if err := client.Identify(&analytics.Identify{
-			UserId: p.API.GetDiagnosticId(),
-		}); err != nil {
-			return err
-		}
-	}
+	client, err := telemetry.NewRudderClient()
 
 	p.client = client
+	return err
+}
 
-	return nil
+func (p *Plugin) initTracker() {
+	enableDiagnostics := false
+	if config := p.API.GetConfig(); config != nil {
+		if configValue := config.LogSettings.EnableDiagnostics; configValue != nil {
+			enableDiagnostics = *configValue
+		}
+	}
+	logger := logger.NewLogger(logger.Config{}, p.API, nil, "")
+	p.tracker = telemetry.NewTracker(p.client, p.API.GetDiagnosticId(), p.API.GetServerVersion(), manifest.Id, manifest.Version, "nps", enableDiagnostics, logger)
```

Comment:
Out of scope of this PR: This API is not easy to use. Passing all this stuff is not easy to set up. Maybe passing a  `mattermost-plugin-api.Client` is the right way.

Reply:
Can be done, yes. Also, I am thinking about dropping the logger here, and just return the error, and let the consumer log or not the error if they want.